Xwt/Xwt.Drawing/Image.cs

```diff
@@ -252,7 +252,7 @@ static bool ParseImageHints (string baseName, string fileName, string ext, out i
 			tags = ImageTagSet.Empty;
 			var firstDelimiter = fileName.IndexOfAny (tagDelimiters);
 
-			if (firstDelimiter <= 0 || fileName.Length <= baseName.Length + 1 || !fileName.StartsWith (baseName, StringComparison.Ordinal))
+			if (firstDelimiter <= 0 || fileName.Length <= baseName.Length + 1 || !fileName.Substring(0, firstDelimiter).Equals(baseName, StringComparison.Ordinal))
```

Comment:
This introduces an allocation here which is probably costly, especially considering how often this is called.

We probably want some other way to validate this.